src/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -249,7 +290,14 @@ const ToggleSwitch = React.forwardRef<HTMLButtonElement, React.PropsWithChildren
         sx={sxProp}
         {...rest}
       >
-        {loading ? <Spinner size="small" /> : null}
+        {isLoadingLabelVisible ? (
+          <VisuallyHidden>
+            <span aria-live="polite" id={loadingLabelId}>
+              Loading
+            </span>
+          </VisuallyHidden>
+        ) : null}
+        {loading ? <Spinner size="small" aria-describedby={loadingLabelId} /> : null}
```

Comment:
Can we move the `aria-describedby={loadingLabelId}` onto the `SwitchButton` instead? Since we're removing the label off of the icon, let's add `aria-hidden` to `<Spinner>`.